dimscord/objects.nim

```diff
@@ -577,74 +618,55 @@ proc newEmoji*(data: JsonNode): Emoji =
-        result.url = some(data["url"].str)
     
     if data.hasKey("timestamps"):
-        result.timestamps = some((start: toBiggestInt(0), final: toBiggestInt(0)))
-        if data["timestamps"].hasKey("start"):
-            get(result.timestamps).start = data["timestamps"]["start"].num
-        if data["timestamps"].hasKey("end"):
-            get(result.timestamps).final = data["timestamps"]["end"].num
+        result.timestamps = some((
+            start: data["timestamps"]{"start"}.getInt.BiggestInt,
+            final: data["timestamps"]{"end"}.getInt.BiggestInt
+        ))
     
-    if data.hasKey("application_id"):
-        result.application_id = some(data["application_id"].str)
-    if data.hasKey("details") and data["details"].kind != JNull:
-        result.details = some(data["details"].str)
-    if data.hasKey("state") and data["state"].kind != JNull:
-        result.state = some(data["state"].str)
-    if data.hasKey("emoji"):
-        result.emoji = some(newEmoji(data["emoji"]))
+    result.assignFrom(data):
+        url <<- some(it.str)
+        application_id <- some(it.str)
+        details <<- some(it.str)
+        state <<- some(it.str)
+        emoji <- some(newEmoji(it))
+        instance <- it.bval
+        flags <- it.getInt()
+
     if data.hasKey("party"):
-        result.party = some((id: "", size: ""))
-        if data["party"].hasKey("id"):
-            get(result.party).id = data["party"]["id"].str
-        if data["party"].hasKey("size"):
-            # TODO: Parse MaxSize too?
-            get(result.party).size = $data["party"]["size"][0].getInt()
+        result.party = some((
+            id: data["party"]{"id"}.getStr,
+            size: data["party"]{"size"}.getStr
+        ))
 
     if data.hasKey("assets"):
         result.assets = some(GameAssets())
-        if data["assets"].hasKey("small_text"):
-            get(result.assets).small_text = data["assets"]["small_text"].str
-        if data["assets"].hasKey("small_image"):
-            get(result.assets).small_image = data["assets"]["small_image"].str
-        if data["assets"].hasKey("large_text"):
-            get(result.assets).large_text = data["assets"]["large_text"].str
-        if data["assets"].hasKey("large_image"):
-            get(result.assets).large_image = data["assets"]["large_image"].str
+        get(result.assets).assignFrom(data["assets"]):
+            small_text <- it.str
+            small_image <- it.str
+            large_text <- it.str
+            large_image <- it.str
 
     if data.hasKey("secrets"):
-        result.secrets = some((join: "", spectate: "", match: ""))
-        if data["secrets"].hasKey("join"):
-            get(result.secrets).join = data["secrets"].str
-        if data["secrets"].hasKey("spectate"):
-            get(result.secrets).spectate = data["spectate"].str
-        if data["secrets"].hasKey("match"):
-            get(result.secrets).match = data["match"].str

```

Comment:
Seemed like there were typos here? `data["secrets"]["join"]`, `data["secrets"]["spectate"]`, and `data["secrets"]["match"]` match up with the predicates. These are assumed in my changes.